src/hyperlink/_url.py

```diff
@@ -749,37 +775,46 @@ class URL(object):
     constructor arguments is below.
 
     Args:
-       scheme (unicode): The text name of the scheme.
-       host (unicode): The host portion of the network location
-       port (int): The port part of the network location. If
-          ``None`` or no port is passed, the port will default to
-          the default port of the scheme, if it is known. See the
-          ``SCHEME_PORT_MAP`` and :func:`register_default_port`
-          for more info.
-       path (tuple): A tuple of strings representing the
-          slash-separated parts of the path.
-       query (tuple): The query parameters, as a dictionary or
-          as an iterable of key-value pairs.
-       fragment (unicode): The fragment part of the URL.
-       rooted (bool): Whether or not the path begins with a slash.
-       userinfo (unicode): The username or colon-separated
-          username:password pair.
-       uses_netloc (bool): Indicates whether two slashes appear
-          between the scheme and the host (``http://eg.com`` vs
-          ``mailto:e@g.com``). Set automatically based on scheme.
-
-    All of these parts are also exposed as read-only attributes of
-    URL instances, along with several useful methods.
+        scheme: The text name of the scheme.
+        host: The host portion of the network location
+        port: The port part of the network location. If ``None`` or no port is
+            passed, the port will default to the default port of the scheme, if
+            it is known. See the ``SCHEME_PORT_MAP`` and
+            :func:`register_default_port` for more info.
+        path: A tuple of strings representing the slash-separated parts of the
+            path.
+        query: The query parameters, as a dictionary or as an iterable of
+            key-value pairs.
+        fragment: The fragment part of the URL.
+        rooted: Whether or not the path begins with a slash.
+        userinfo: The username or colon-separated username:password pair.
+        uses_netloc: Indicates whether two slashes appear between the scheme
+            and the host (``http://eg.com`` vs. ``mailto:e@g.com``).
+            Set automatically based on scheme.
+
+    All of these parts are also exposed as read-only attributes of URL
+    instances, along with several useful methods.
```

Comment:
I removed the type information from the docstring, since (a) some of it is actually inaccurate and (b) it's redundant given the type hints.

Reply:
Maybe we should upgrade/setup hyperlink's sphinx to include those in the docs somehow? (This is [Napoleon doing Google-style docs](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html), btw.)

Reply:
I don't know enough about Sphinx to do that, but I agree that it would be cool if it could be set up that way.

Reply:
Sure, I'll take a swing at that, then. Gotta do something around here, after all! :)

Reply:
:-) no worries; we pitch in when we have cycles and I just happen to have a bust of energy for this right now.  I haven't even gotten to the think I want to work on yet!

But if you can figure out how to get Sphinx to show the type hints, that would really be awesome.  If not, I thing I can go through and put these back with more accurate (but, sadly, duplicate) info.

Reply:
https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#type-annotations

Reply:
ARGH! "Python 2/3 compatible annotations aren’t currently supported by Sphinx and won’t show up in the docs."